src/prisma/cli/_node.py

```diff
@@ -0,0 +1,320 @@
+from __future__ import annotations
+
+import os
+import sys
+import shutil
+import logging
+import subprocess
+from pathlib import Path
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, IO, Union, Any, Mapping, cast
+from typing_extensions import Literal
+
+from pydantic.typing import get_args
+
+from .. import config
+from .._proxy import LazyProxy
+from ..binaries import platform
+from ..errors import PrismaError
+
+if TYPE_CHECKING:
+    # TODO
+    import nodejs  # type: ignore
+    import nodejs.node  # type: ignore
+    import nodejs.npm  # type: ignore
+else:
+    try:
+        import nodejs
+    except ImportError:
+        nodejs = None
+
+
+log: logging.Logger = logging.getLogger(__name__)
+File = Union[int, IO[Any]]
+Target = Literal['node', 'npm']
+
+
+class UnknownTargetError(PrismaError):
+    def __init__(self, *, target: str) -> None:
+        super().__init__(
+            f'Unknown target: {target}; Valid choices are: {", ".join(get_args(cast(type, Target)))}'
+        )
+
+
+class Strategy(ABC):
+    resolver: Literal['nodejs-bin', 'global', 'nodeenv']
+
+    # TODO: support more options
+    def run(
+        self,
+        *args: str,
+        check: bool = False,
+        cwd: Path | None = None,
+        stdout: File | None = None,
+        stderr: File | None = None,
+        env: Mapping[str, str] | None = None,
+    ) -> subprocess.CompletedProcess[bytes]:
+        """Call the underlying Node.js binary.
+
+        The interface for this function is very similar to `subprocess.run()`.
+        """
+        return self.__run__(
+            *args,
+            check=check,
+            cwd=cwd,
+            stdout=stdout,
+            stderr=stderr,
+            env=_update_path_env(
```

Comment:
Can you move this up and print it at the debug level? Might help us from a logging standpoint understand the CI failure -- https://github.com/RobertCraigie/prisma-client-py/actions/runs/3400416165/jobs/5654829166#step:3:1336

Reply:
I think a better solution would be to add debug logging directly in the `_update_path_env` function - thanks for pointing out that this needs better logging :)

Also I think the issue is that I had used `:` as the path separator when windows actually uses `;`, hopefully that's fixed now.

Reply:
Yeah I saw that it was a good catch. Regardless of where you add it, will be useful in bug reports we get